drivers/led/ltc3220/ltc3220.c

```diff
@@ -0,0 +1,537 @@
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include <errno.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include "ltc3220.h"
+#include "no_os_i2c.h"
+#include "no_os_util.h"
+#include "no_os_delay.h"
+
+
+/***************************************************************************//**
+ * @brief Initializes the LTC3220 device structure.
+ * @param device - The device structure to initialize.
+ * @param init_param - The initialization parameters.
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc3220_init(struct ltc3220_dev **device,
+		 struct ltc3220_init_param init_param)
+{
+	struct ltc3220_dev *dev;
+	int ret;
+
+	dev = (struct ltc3220_dev *)no_os_calloc(1, sizeof(*dev));
+	if (!dev)
+		return -ENOMEM;
+
+	ret = no_os_i2c_init(&dev->i2c_desc, &init_param.i2c_init_param);
+	if (ret)
+		goto ltc3220_init_err;
+
+	ret = no_os_gpio_get(&dev->gpio_rst_desc, &init_param.gpio_init_param);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_direction_output(dev->gpio_rst_desc, NO_OS_GPIO_HIGH);
+	if (ret)
+		return ret;
+
+	*device = dev;
+
+	return 0;
+
+ltc3220_init_err:
+	ltc3220_remove(dev);
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Deallocates the resources for the device structure.
+ * @param device - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc3220_remove(struct ltc3220_dev *device)
+{
+	int ret;
+
+	if (!device)
+		return -ENODEV;
+
+	if (device->i2c_desc) {
+		ret = no_os_i2c_remove(device->i2c_desc);
+		if (ret)
+			return ret;
+		device->i2c_desc = NULL;
+	}
+
+	if (device->gpio_rst_desc) {
+		ret = no_os_gpio_remove(device->gpio_rst_desc);
+		if (ret)
+			return ret;
+		device->gpio_rst_desc = NULL;
+	}
+
+	no_os_free(device);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Resets the device using the RST pin
+ * @param device - the device structure.
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int ltc3220_reset(struct ltc3220_dev *device)
+{
+	int ret;
```

Comment:
variable declaration part should be separated by the rest of the code by an empty line.